crates/core/src/db/queries/collections.rs

```diff
@@ -235,3 +235,113 @@ pub fn collection_activities(
         .load(conn)
         .context("Failed to load collection activities")
 }
+
+const LISTED_COUNT_QUERY: &str = r"
+SELECT COUNT(listings.id) AS value
+FROM listings
+INNER JOIN metadatas ON(listings.metadata = metadatas.address)
+INNER JOIN metadata_collection_keys ON(metadatas.address = metadata_collection_keys.metadata_address)
+INNER JOIN auction_houses ON(listings.auction_house = auction_houses.address)
+WHERE
+    metadata_collection_keys.collection_address = $1
+	AND auction_houses.treasury_mint = 'So11111111111111111111111111111111111111112'
+	AND listings.purchase_id IS NULL
+	AND listings.canceled_at IS NULL
+	AND metadata_collection_keys.verified = true
+GROUP BY metadata_collection_keys.collection_address;
+
+-- $1: address::text";
+
+/// Computes the count of active listings of NFTs in the collection.
+///
+/// # Errors
+/// This function fails if the underlying SQL query returns an error
+pub fn listed_count(conn: &Connection, address: impl ToSql<Text, Pg>) -> Result<i64> {
+    let query_result = diesel::sql_query(LISTED_COUNT_QUERY)
+        .bind(address)
+        .load::<BigIntWrapper>(conn)
+        .context("Failed to load collection listing count")?;
+
+    query_result
+        .get(0)
+        .ok_or_else(|| anyhow!("Collection not found."))
+        .map(|w| w.value)
+}
+
+const VOLUME_TOTAL_QUERY: &str = r"
+SELECT CAST(SUM(purchases.price) AS BIGINT) AS value
+FROM purchases
+INNER JOIN metadatas ON(purchases.metadata = metadatas.address)
+INNER JOIN metadata_collection_keys ON(metadatas.address = metadata_collection_keys.metadata_address)
+INNER JOIN auction_houses ON(purchases.auction_house = auction_houses.address)
+WHERE
+    metadata_collection_keys.collection_address = $1
+    AND purchases.created_at >= $2
+    AND purchases.created_at <= $3
```

Comment:
Appreciate the flexibility of setting the time frame but to simplify the requirements for the client I recommend we drop the start and end date range and return all purchases since we are using for total volume.

Reply:
aye aye; it was part of the ticket, but I will remove